src/Features/Core/Portable/Navigation/IDocumentNavigationService.cs

```diff
@@ -15,31 +16,46 @@ internal interface IDocumentNavigationService : IWorkspaceService
         /// <summary>
         /// Determines whether it is possible to navigate to the given position in the specified document.
         /// </summary>
-        bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan);
+        bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, CancellationToken cancellationToken);
 
         /// <summary>
         /// Determines whether it is possible to navigate to the given line/offset in the specified document.
         /// </summary>
-        bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset);
+        bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset, CancellationToken cancellationToken);
 
         /// <summary>
         /// Determines whether it is possible to navigate to the given virtual position in the specified document.
         /// </summary>
-        bool CanNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace = 0);
+        bool CanNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace, CancellationToken cancellationToken);
 
         /// <summary>
         /// Navigates to the given position in the specified document, opening it if necessary.
         /// </summary>
-        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options = null);
+        bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options, CancellationToken cancellationToken);
```

Comment:
There's a pretty good chance some of these methods may be used via IVT -- please do test prior to merging.

Reply:
er.  What sorts of other IVT do we have?  I already updated TS/F# based on the EA layers :)

Reply:
Those are the main ones...maybe some Razor stuff?

Reply:
It looks like it'll impact VS for Mac but that should be a trivial impact. FYI to @davidwengier.